pages/mintKamon/index.tsx

```diff
@@ -48,26 +48,30 @@ const MintKamon: NextPage = () => {
 
   // approve
   const { status, approve } = useApprove(henkakuErc20, kamonNFT)
-  const approved = useApproval(henkakuErc20, kamonNFT)
+  const approved = useApproval(henkakuErc20, kamonNFT, data?.address)
 
   // mint
   const mintPrice = 1000
-  const { isMinting, mint } = useMintWithHenkaku(tokenURI, mintPrice)
+  const { isMinting, mint } = useMintWithHenkaku(kamonNFT, tokenURI, mintPrice)
   const mintWithHenkaku = useCallback(async () => {
     const data = await mint()
     if (data) {
       setHasNFT(true)
     }
   }, [mint, setHasNFT])
-  const { balanceOf } = useBalanceOf(data?.address)
-  const { totalSupply } = useTotalSupply()
+  const { balanceOf } = useBalanceOf(kamonNFT, data?.address)
+  const { totalSupply } = useTotalSupply(kamonNFT)
 
   useEffect(() => {
-    setHasNFT(balanceOf && balanceOf.gte(1) > 0 ? true : false)
+    if (data?.address) {
+      setHasNFT(balanceOf && balanceOf.gte(1) > 0 ? true : false)
+    }
   }, [])
```

Comment:
If you are not implementing this with the intention of only running it when mounted and unmounted only, you may want to add it.

```suggestion
  }, [balanceOf, data?.address])
```

Reply:
Thanks for your comment! It is very helpful. I think you are right, and I will correct this along with the above.